nntrainer/opencl/opencl_buffer.hpp

```diff
@@ -0,0 +1,45 @@
+// SPDX-License-Identifier: Apache-2.0
+/**
+ * Copyright (C) 2023 Debadri Samaddar <s.debadri@samsung.com>
+ *
+ * @file    opencl_buffer.hpp
+ * @date    06 Feb 2024
+ * @see     https://github.com/nnstreamer/nntrainer
+ * @author  Debadri Samaddar <s.debadri@samsung.com>
+ * @bug     No known bugs except for NYI items
+ * @brief   OpenCL wrapper for buffer usage
+ *
+ */
+
+#ifndef GPU_CL_OPENCL_BUFFER_HPP_
+#define GPU_CL_OPENCL_BUFFER_HPP_
+
+#include "opencl_command_queue_manager.hpp"
+#include "opencl_context_manager.hpp"
+#include "third_party/cl.h"
+
+namespace nntrainer::internal {
```

Comment:
How about adding more comments following the oxygen to provide a better understanding?